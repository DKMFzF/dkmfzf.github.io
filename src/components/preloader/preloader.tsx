import { FC } from 'react';
import { PreloaderUI } from '../ui';

export const Preloader: FC = () => <PreloaderUI />;
