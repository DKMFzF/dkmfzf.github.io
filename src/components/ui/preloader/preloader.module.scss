.preloader {
	align-items: center;
	display: flex;
	flex-grow: 1;
	justify-content: center;
}

.preloader-circle {
	animation: spin 0.75s infinite linear;
	border: 1px solid;
	border-color: #d1d2d6 #9fa0a5 #626368 #1a1b22;
	border-radius: 50%;
	display: block;
	height: 74px;
	inset: 0;
	margin: auto;
	position: relative;
	width: 74px;
}

@keyframes spin {
	100% {
		transform: rotate(360deg);
	}
}
