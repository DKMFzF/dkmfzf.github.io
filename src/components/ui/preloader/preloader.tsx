import styles from './preloader.module.scss';

export const PreloaderUI = () => (
	<div className={styles.preloader}>
		<div className={styles['preloader-circle']} />
	</div>
);
