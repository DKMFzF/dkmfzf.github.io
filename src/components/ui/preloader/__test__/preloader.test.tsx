import { testRenderComponent } from '@utils-testing';
import { PreloaderUI } from '../preloader';

describe('[PreloaderUI]', () => {
	it('Renders preloader correctly and matches snapshot', () =>
		testRenderComponent(<PreloaderUI />));
});
