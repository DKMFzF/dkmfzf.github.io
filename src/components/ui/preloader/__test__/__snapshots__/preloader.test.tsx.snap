// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[PreloaderUI] Renders preloader correctly and matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="preloader"
  >
    <div
      class="preloader-circle"
    />
  </div>
</DocumentFragment>
`;
